Dshop/apps/users/api_views.py

```diff
@@ -0,0 +1,48 @@
+from django.contrib.auth.models import User
+from django.db import IntegrityError
+from django.utils.translation import gettext_lazy as _
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+# from rest_framework.authtoken.models import Token
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import CustomUser
+from .serializers import RegistrationSerializer
+
+
+class RegistrationView(APIView):
+    """
+    API View responsible for single user registration
+
+    :accepts: default Django User model fields (django.contrib.auth.models.User) [username, email, password1, password2]
+    :returns: 201 / 400
+    """
+    permission_classes = (AllowAny,)
+    serializer_class = RegistrationSerializer
+    throttle_scope = 'registration'
+
+    def post(self, request):
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        cleaned_data = serializer.get_cleaned_data()
+
+        # Try to create new object instance with the validated data of username and email
+        try:
+            user = User.objects.create_user(
+                username=cleaned_data['username'],
+                email=cleaned_data['email'],
+                password=cleaned_data['password']
+            )
+        except IntegrityError:
+            user = None
+
+        if user:
+            # finish registration

```

Comment:
Please remove comment. The code is self-explanatory. Or explain that is needed because we have 1:1 relationship with django user. 
In general, comments are bad because nobody updates those later on. And if there is comment, there is something wrong with code in most cases.

The only good case for comment is to explain not what the code does (create customuser or finish registration) but why it is doing it this way. Example: 
```
sleep(5) # we have a lousy external service that needs 3 seconds to process short requests so adding 5 delay just in case
```
Do you see difference?